controller/src/bin/xcvradm.rs

```diff
@@ -632,6 +836,109 @@ fn load_known_write_data<R: Read>(rdr: &mut R, kind: InputKind) -> anyhow::Resul
     }
 }
 
+#[derive(Clone, ValueEnum, PartialEq)]
+enum StatusFields {
+    /// The port number of the switch.
+    Port,
+    /// The power state of the transceiver module.
+    Status,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum PowerFields {
+    /// The port number of the switch.
+    Port,
+    /// The power state of the transceiver module.
+    Power,
+    /// Flag to indicate a software override of the modules power state.
+    Override,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum IdentifyFields {
+    /// The port number of the switch.
+    Port,
+    /// SFF-8024 identifier of the transceiver module.
+    Ident,
+    /// ??? - Needs details
+    Description,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum VendorInfoFields {
+    /// The port number of the switch.
+    Port,
+    /// Identifier of the transceiver module.
+    Identifier,
+    /// Manufacturer of the transceiver module.
+    Vendor,
+    /// Manufacturer part number.
+    Part,
+    /// Revision number of the transceiver module.
+    Rev,
+    /// Serial number of the transceiver module.
+    Serial,
+    /// ??? - Needs details
+    ManufacturedDate,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum ReadDataFields {
+    /// The port number of the switch.
+    Port,
+    /// The requested data from the transceiver module.
+    Data,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum ReadUpperFields {}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum MemoryModelFields {
+    /// The port number of the switch.
+    Port,
+    /// The model number of the transceiver module.
+    Model,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum MacFields {
+    /// The port number of the switch.
+    Port,
+    /// The model number of the transceiver module.
+    Model,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum LedFields {
+    /// The port number of the switch.
+    Port,
+    /// State of th transceiver modules status LED.
+    Led,
+}
+
+#[derive(Clone, ValueEnum, PartialEq)]
+enum MonitorFields {
+    /// The port number of the switch.
+    Port,
+    /// The temperature of the transceiver module.
+    Temperature,
+    /// ??? - Is this power being supplied to the module?
+    SupplyVoltage,
+    /// Average received power of the transceiver module in mW.
+    AverageRxPower,
+    /// Transmit bias current of the transceiver module in mA.
+    TxBias,
+    /// Transmit power of the transceiver module in mW.
+    TxPower,
+    /// ???
```

Comment:
These should just be described as "Auxiliary monitor N". There are a bunch of things modules _can_ report in here, but it's entirely part-specific. Examples include the laser frequency, or information about the cooling system. We can't know what it's describing until we read it from the module.